.github/workflows/e2e_runtime-reproducibility.yml

```diff
@@ -64,25 +71,34 @@ jobs:
     permissions:
       contents: read
     needs: os-matrix
+    strategy:
+      matrix:
+        build-target: ["hello"]
     steps:
+      - name: Download all checksum artifacts
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          pattern: "${{matrix.build-target}}-*-checksums"
+          path: "./checksums"
+          merge-multiple: true
       - name: Collect checksums
         id: collect
         shell: python
         run: |
-          import json, os
+          import json, os, pathlib
+          checksum_mismatch = False
           seen = {}
-          with open(os.getenv("GITHUB_OUTPUT")) as f:
-              for line in f:
-                  matrix, checksum = line.strip().split("=")
-                  if not checksum in seen:
-                      seen[checksum] = []
-                  seen[checksum].append(matrix)
-
+          for file in os.listdir("./checksums"):
+              checksum = pathlib.Path("./checksums", file).read_text()
+              if not checksum in seen:
+                  seen[checksum] = []
```

Comment:
```suggestion
          seen = { pathlib.Path("./checksums", file).read_text() for file in os.listdir("./checksums") }
```

Reply:
forgot to add info where checksum was seen, added it. List comprehensions like this aren't possible with the new change.